src/tribol/search/InterfacePairFinder.cpp

```diff
@@ -1,643 +1,873 @@
+        IndexT toIdx = i % mesh2NumElems;
+        if (is_symm)
+        {
+          IndexT row = algorithm::symmMatrixRow(i, mesh1NumElems);
+          IndexT offset = row * (row + 1) / 2;
+          fromIdx = row;
+          toIdx = i - offset;
+        }
+
+        // get unique index for the array
+        auto idx = RAJA::atomicInc<RAJA::auto_atomic>(pCount);
+
+        pairs_view[idx] = InterfacePair(fromIdx, toIdx, true);
+      }
+    );
 
+    SLIC_INFO("Coupling scheme has " << contactPairs.size()
+          << " pairs out of a maximum possible of " << numPairs
+          << " = " << mesh1NumElems << " * " << mesh2NumElems << ".");
+  }
 private:
-   const MeshData* m_meshData;
+  CouplingScheme* m_coupling_scheme;
+};  // End of CartesianProduct definition
 
-   VertSet m_vertSet;
-   ElemSet m_elemSet;
-   ElemVertRelation m_elemVertConnectivity;
-};
+///////////////////////////////////////////////////////////////////////////////
 
 /*!
- * \brief Helper class to compute the candidate pairs for a coupling scheme
+ * \brief Implicit Grid helper class to compute the candidate pairs for a coupling scheme
  *
  * A GridSearch indexes the elements from the first mesh of
  * the coupling scheme in a spatial index that requires element bounding boxes.
  * Then, for each of the elements in the second mesh, we find the candidate
  * pairs and add them to the coupling scheme's list of contact pairs.
  *
- * The spatial index is generated in \a generateSpatialIndex()
+ * The spatial index is generated in \a initialize()
  * and the search is performed in \a findInterfacePairs()
  *
  * \tparam D The spatial dimension of the coupling scheme mesh vertices.
  */
 template<int D>
-class GridSearch
+class GridSearch : public SearchBase
 {
 public:
-   using ImplicitGridType = spin::ImplicitGrid<D, axom::SEQ_EXEC, int>;
-   using SpacePoint = typename ImplicitGridType::SpacePoint;
-   using SpaceVec = typename ImplicitGridType::SpaceVec;
-   using SpatialBoundingBox = typename ImplicitGridType::SpatialBoundingBox;
-
-   /*!
-    * Constructs a GridSearch instance over CouplingScheme \a couplingScheme
-    * \pre couplingScheme is not null
-    */
-   GridSearch(CouplingScheme* couplingScheme)
-      : m_couplingScheme(couplingScheme)
-   {
-      MeshManager & meshManager = MeshManager::getInstance();
+  using BBox = primal::BoundingBox<RealT, D>;
+  using PointT = primal::Point<RealT, D>;
+
+  using ImplicitGridType = spin::ImplicitGrid<D,axom::SEQ_EXEC,int>;
+  using SpacePoint = typename ImplicitGridType::SpacePoint;
+  using SpaceVec = typename ImplicitGridType::SpaceVec;
+  using SpatialBoundingBox = typename ImplicitGridType::SpatialBoundingBox;
+
+  /*!
+   * Constructs a GridSearch instance over CouplingScheme \a couplingScheme
+   * \pre couplingScheme is not null
+   */
+  GridSearch( CouplingScheme* couplingScheme )
+    : m_coupling_scheme( couplingScheme )
+    , m_mesh1( m_coupling_scheme->getMesh1() )
```

Comment:
This will need to update if `getMesh1()` returns the meshData and not the view like we discussed. I see here how you want the view though.